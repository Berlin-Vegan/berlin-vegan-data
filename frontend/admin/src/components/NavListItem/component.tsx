import React, { FunctionComponent } from 'react';
import { NavLink, useLocation } from 'react-router-dom';
import ListItem from '@material-ui/core/ListItem';

type NavListItemProps = {
  pathname: string;
};

const NavListItem: FunctionComponent<NavListItemProps> = ({
  children,
  pathname,
}) => {
  const location = useLocation();

  return (
    <ListItem
      button
      component={NavLink}
      to={pathname}
      selected={location.pathname === pathname}
    >
      {children}
    </ListItem>
  );
};

export default NavListItem;
