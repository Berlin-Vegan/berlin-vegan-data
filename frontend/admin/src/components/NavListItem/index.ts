import NavListItem from './component';

export default NavListItem;
