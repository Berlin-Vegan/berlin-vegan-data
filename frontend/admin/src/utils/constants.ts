export const GASTRO: string = 'GASTRO';
export const SHOPPING: string = 'SHOPPING';

export type LocationType = typeof GASTRO | typeof SHOPPING;

const API = '/api/';
export const V1 = `${API}v1/`;
export const V2 = `${API}v2/`;
