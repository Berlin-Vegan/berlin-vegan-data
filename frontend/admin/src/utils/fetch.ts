import { getCSRFToken } from './cookie';
import { UserDispatch, TYPE_USER_LOGOUT } from '../providers/UserProvider';

export const authorizedFetch = async (
  userDispatch: UserDispatch,
  url: string,
  method: string = 'GET',
  body: object | Array<unknown> | null = null
): Promise<Response> => {
  const response = await fetch(url, {
    method,
    body: body ? JSON.stringify(body) : body,
    headers: {
      'Content-Type': 'application/json',
      'X-CSRFToken': getCSRFToken(),
    },
  });

  if (response.status === 403) {
    userDispatch({ type: TYPE_USER_LOGOUT });
  }

  return response;
};

export const fetchGastroList = async (
  userDispatch: UserDispatch,
  filter: string
): Promise<Response> =>
  authorizedFetch(userDispatch, `/api/v1/gastros/?${filter}`);
