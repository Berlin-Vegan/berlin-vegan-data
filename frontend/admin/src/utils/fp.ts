import {
  addIndex,
  compose,
  curry,
  defaultTo,
  join,
  map,
  nth,
  pipe,
  splitAt,
  toLower,
  toUpper,
} from 'ramda';

export const nthOr = curry((defaultVal, list) =>
  compose(defaultTo(defaultVal), nth)(list),
);

export const nthOrEmpty = nthOr('');

export const mapIndexed = addIndex(map);

export const capitalizeWord = pipe(
  toLower,
  splitAt(1),
  (splitted) => [toUpper(nthOrEmpty(0)(splitted)), nthOrEmpty(1)(splitted)],
  join(''),
);
