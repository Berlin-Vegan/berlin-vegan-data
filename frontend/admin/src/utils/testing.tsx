import renderer from 'react-test-renderer';
import React, { ComponentType } from 'react';
import ReactDOM from 'react-dom';
import { act } from 'react-dom/test-utils';

export const standardComponentTest = (C: ComponentType<any>, props = {}) => {
  renderWithTestRenderer(C, props);
  it('renders without crashing', () => renders(C, props));
};

export const standardComponentTestMockFetch = (
  C: ComponentType<any>,
  props = {},
  fakeResponse: any,
) => {
  renderWithTestRenderer(C, props);
  it('renders without crashing and mock fetch', async () => {
    jest.spyOn(window, 'fetch').mockImplementation((): Promise<any> => {
      const fetchResponse = {
        json: () => Promise.resolve(fakeResponse),
      };
      return Promise.resolve(fetchResponse);
    });
    await act(async () => renders(C, props));
  });
};

export const renders = (C: ComponentType, props: object) => {
  const div = document.createElement('div');
  ReactDOM.render(<C {...props} />, div);
};

const getName = (C: ComponentType) => C.displayName || C.name || 'A Component';

const renderWithTestRenderer = (C: ComponentType, props: object) => {
  it(`render <${getName(C)} />`, () => {
    const tree = renderer.create(<C {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
};
