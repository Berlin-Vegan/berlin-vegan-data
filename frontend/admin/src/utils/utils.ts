import { pipe, toLower } from 'ramda';
import { LocationType, V1, V2 } from './constants';

//
export const buildReviewListUrl: string = `${V1}review/`;
export const buildReviewDetailUrl = (id: number): string =>
  `${buildReviewListUrl}${id}/`;

// API V2 urls
export const buildListUrl = (type: LocationType): string =>
  pipe(toLower, (lowerType: string) => `${V2}${lowerType}/`)(type);
export const buildDetailUrl = (type: LocationType, id: string): string =>
  pipe(toLower, buildListUrl, (baseUrl: string) => `${baseUrl}${id}/`)(type);
// FE urls
export const buildFEDetailUrl = (type: LocationType, id: string): string =>
  pipe(toLower, (lowerType: string) => `/${lowerType}/${id}/edit`)(type);
