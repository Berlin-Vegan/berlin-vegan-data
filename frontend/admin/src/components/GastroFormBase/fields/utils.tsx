import React from 'react';
import MenuItem from '@material-ui/core/MenuItem';
import { map } from 'ramda';
import { nthOr } from '../../../utils/fp';

const menuItem = (value: number | string, label: string) => (
  <MenuItem value={value} key={value}>
    {label}
  </MenuItem>
);

export const menuItems = (fieldOptions: Array<[number | string, string]>) =>
  map((item) => menuItem(nthOr('', 0)(item), nthOr('', 1)(item)))(fieldOptions);

export const withLeadingZero = (value: number): string =>
  value < 10 ? `0${value}` : `${value}`;
