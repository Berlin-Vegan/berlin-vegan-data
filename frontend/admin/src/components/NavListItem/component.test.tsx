import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { standardComponentTest } from '../../utils/testing';
import NavListItem from './component';

const NavListItemWrapper = () => (
  <MemoryRouter>
    <NavListItem pathname="test" />
  </MemoryRouter>
);

standardComponentTest(NavListItemWrapper);
