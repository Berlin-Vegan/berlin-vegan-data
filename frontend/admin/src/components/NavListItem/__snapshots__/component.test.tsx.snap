// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render <NavListItemWrapper /> 1`] = `
<a
  aria-current={null}
  aria-disabled={false}
  className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
  href="/test"
  onBlur={[Function]}
  onClick={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  role="button"
  tabIndex={0}
/>
`;
